src/main/java/com/github/manolo8/darkbot/backpage/DispatchManager.java

```diff
@@ -0,0 +1,172 @@
+                    .collect(Collectors.toList()));
+        }
+
+        private boolean update(String page, DispatchData data) {
+            Matcher m = regex.matcher(page);
+            if (!m.find()) return false;
+
+            do {
+                int max = Math.min(m.groupCount(), consumers.size());
+                for (int i = 0; i < max; i++)
+                    consumers.get(i).accept(data, m.group(i + 1));
+            } while (m.find());
+
+            return m.groupCount() == consumers.size();
+        }
+
+        public static boolean updateAll(String page, DispatchData data) {
+            boolean updated = true;
+            for (InfoReader reader : InfoReader.values()) {
+                updated &= reader.update(page, data);
+            }
+            return updated;
+        }
+    }
+
+    public boolean hireRetriever(Retriever retriever){
+        if(data.getAvailableSlots() > 0) {
+            //retriever.getCost(); // TODO: check cost
+            try {
+                String response = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
+                        .setRawParam("command", "sendDispatch")
+                        .setRawParam("dispatchId", retriever.getId())
+                        .getContent();
+                if(response.contains("\"result\":\"ERROR\"")){
+                    System.out.println("Failed to dispatch " + retriever.getId() + ": " + response);
+                    update(10_000);
+                }else{
+                    System.out.println("Successfully dispatched " + retriever.getId() + ": " + response);
+                    update(10_000);
+                    return true;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+                System.out.println("Exception hiring dispatcher: " + e);
+            }
+        }
+        return false;
+    }
+
+    public boolean collect(InProgress progress) {
+        try {
+            if(progress.getCollectable().equals("0")) return false;
+            System.out.println("Collecting: Slot " + progress.getSlotID());
+            String x = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
+                    .setRawParam("command", "collectDispatch")
+                    .setRawParam("slot", progress.getSlotID())
+                    .getContent();
+
+            //parse response
+            //{"result":"OK","message":"Collected the following:","rewardsLog":[{"lootId":"Solidus","amount":3},{"lootId":"PLT-2026","amount":97},{"lootId":"Scrap","amount":3}]}
+
+            if (x.contains("\"result\":\"ERROR\"")) {
+                System.out.println("Unable to collect retriever");
+            } else {
+                System.out.println("Dispatch Collected: " + progress.getSlotID() + " : " + x.substring(x.indexOf("rewardsLog")));
+                return true;
+            }
+            update(1000);
```

Comment:
this update will never be reached on error, which can make something stuck in a forever-collect loop, put it in both parts of the if and put it with an update -1, like above.

As i said in the previous review, consider moving this whole if-else to a private method:

```java
    public boolean handleResponse(String type, String id, String response) {
        boolean failed = response.contains("\"result\":\"ERROR\"")
        System.out.println(type + " (" + id + ") " + (failed ? "failed" : "succeeded") + ": " + response);
        update(-1);
        return !failed;
    }
```

then you can call it from both collect and hire retriever like:

`handleResponse("Dispatch collected", progress.getSlotID(), response);`
`handleResponse("Hired dispatcher", retriever.getId(), response);`